pkg/vm/engine/tae/tables/block.go

```diff
@@ -905,3 +905,56 @@ func (blk *dataBlock) GetSortColumns(schema *catalog.Schema, data *containers.Ba
 	}
 	return vs
 }
+
+func (blk *dataBlock) CollectInsert(start, end types.TS, buffer *bytes.Buffer) *containers.Batch {
```

Comment:
1. CollectInsert should be rename to "CollectAppendsInRange"
2. CollectDelete should be rename to "CollectDeletesInRange"
3. Both CollectXXX should be thread-safe
4. CollectAppendsXXX should process appendable and non-appendable seperately